onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -7292,12 +7292,152 @@ def aten_repeat(self: TTensor, repeats: Sequence[TInt]) -> TTensor:
     return op.Tile(self_expanded, repeats)
 
 
-def aten_repeat_interleave(
-    repeats: TensorType, output_size: Optional[int] = None
+@torch_op("aten::repeat_interleave.self_int", trace_only=True)
+def aten_repeat_interleave_int(
+    self: TensorType, repeats: int, dim: Optional[int]
 ) -> TensorType:
-    """repeat_interleave.Tensor(Tensor repeats, *, int? output_size=None) -> Tensor"""
+    """repeat_interleave.Tensor(Tensor repeats, *, int? output_size=None) -> Tensor
 
-    raise NotImplementedError()
+    The trick is to repeat in one direction orthogonal to reshape.
+
+    .. code-block:: python
+
+        x = torch.tensor([[0, 1, 2], [3, 4, 5]])
+        x.repeat_interleave(2, dim=0)
+
+    is equivalent to:
+
+    .. code-block:: python
+
+        x = torch.tensor([[0, 1, 2], [3, 4, 5]])
+        x.repeat((1, 2)).reshape((-1, t.shape[1]))
+    """
+    if dim is None:
+        raise NotImplementedError("No conversion available yet when dim is None.")
+
+    self_rank = len(self.shape)
+    pos_dim = (dim + self_rank) % self_rank
+    unsqueezed = op.Unsqueeze(self, [pos_dim + 1])
+    onehot = op.Concat(
+        op.ConstantOfShape(
+            op.Constant(value_ints=[self_rank]), value=ir.tensor([1], dtype=INT64.dtype)
+        ),
+        op.Constant(value_ints=[repeats]),
+        axis=0,
+    )
+    tiled = op.Tile(unsqueezed, onehot)
```

Comment:
Shouldn't we be inserting the extra dimension `repeats` just after `pos_dim` (corresponding to the unsqueezed axis above)? Eg., for a 2D tensor with repeats=3 and dim=0, we should use `[1, 3, 1]` wherer the unsqueezed shape is `[Rows, 1, Cols]`?

Reply:
I rewrote everything to make it more simple.